src/lib.rs

```diff
@@ -46,6 +67,200 @@ impl UniffiCustomTypeConverter for Uuid {
     }
 }
 
+pub struct StatusLists;
+
+impl TypedStatusMapProvider<Uri, BitstringStatusListCredential> for StatusLists {
+    async fn get_typed(&self, id: &Uri) -> Result<MaybeCached<StatusList>, ProviderError> {
+        eprintln!("fetch <{id}>");
+        Ok(MaybeCached::NotCached(StatusList::from_bytes(
+            1.try_into().unwrap(),
+            vec![0u8; 125],
+            TimeToLive::DEFAULT,
+        )))
+    }
+}
+
+#[uniffi::export]
+pub async fn verify_pdf417_barcode(payload: String) -> Result<(), VCBVerificationError> {
+    let mut cursor = Cursor::new(payload);
+    let mut file = pdf_417::File::new(&mut cursor).map_err(|e| VCBVerificationError::Generic {
+        value: e.to_string(),
+    })?;
+    let dl: DlSubfile = file
+        .read_subfile(b"DL")
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })?
+        .ok_or(VCBVerificationError::Generic {
+            value: "Invalid DLSubfile".to_string(),
+        })?;
+    let zz: ZZSubfile = file
+        .read_subfile(b"ZZ")
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })?
+        .ok_or(VCBVerificationError::Generic {
+            value: "Invalid ZZSubfile".to_string(),
+        })?;
+    let vc = zz
+        .decode_credential()
+        .await
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })?;
+
+    let status_list_client = ConstTerseStatusListProvider::new(
+        StatusLists,
+        StatusListInfo::new(1000, StatusPurpose::Revocation),
+    );
+
+    let params = VerificationParameters::new_with(
+        AnyDidMethod::default().into_vm_resolver(),
+        status_list_client,
+    );
+
+    verify(&vc, &dl.mandatory, params)
+        .await
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })?
+        .map_err(|e| VCBVerificationError::Generic {
```

Comment:
Could you add a new variant to say "verification failed" -- so that the app can distinguish between something going wrong and the verification simply failing